.github/workflows/CI.yml

```diff
@@ -7,33 +7,33 @@ on:
 
 env:
   NIGHTLY_TEST_SETTINGS: true
-  # By default `smokeTest` will run `make check`, `make mason`, and `make
-  # docs`. We split those into 3 separate jobs to speed up CI times, so
-  # disable all options here, and selectively enable below
-  CHPL_SMOKE_SKIP_MAKE_CHECK: true
-  CHPL_SMOKE_SKIP_MAKE_MASON: true
-  CHPL_SMOKE_SKIP_DOC: true
+  # store the dependencies in one place
+  CHPL_APT_DEPS: gcc g++ m4 perl python3 python3-pip python3-venv python3-dev bash make mawk git pkg-config cmake llvm-11-dev llvm-11 llvm-11-tools clang-11 libclang-11-dev libedit-dev
 
 jobs:
   make_check:
     runs-on: ubuntu-latest
-    env:
-      CHPL_SMOKE_SKIP_MAKE_CHECK: false
     steps:
     - uses: actions/checkout@v2
+    - name: install deps
+      run: sudo apt-get install $CHPL_APT_DEPS
     - name: make check
       run: |
-        CHPL_LLVM=none ./util/buildRelease/smokeTest
+        ./util/buildRelease/smokeTest chpl
 
   make_doc:
     runs-on: ubuntu-latest
-    env:
-      CHPL_SMOKE_SKIP_DOC: false
     steps:
     - uses: actions/checkout@v2
+    - name: install deps
+      run: sudo apt-get install cmake doxygen
+    - name: make libchplcomp-docs
+      run: |
+        make libchplcomp-docs
```

Comment:
Just checking if there's any tradeoffs to putting this here or in the `test-libchplcomp` check. I have no preference just thinking that step is faster if we're trying to cut down total time.

Reply:
Right. Here I'm trying to get the docs build to be complete. This part builds the compiler-next docs so that they will be included in the .zip docs build we can get as an artifact. (If it weren't connected to the rest of the docs build and we just wanted to make sure that make target succeded, it would be better in the test-libchplcomp check).